service/history/execution/context.go

```diff
@@ -260,8 +260,9 @@ func (c *contextImpl) LoadWorkflowExecutionWithTaskVersion(
 
 	if c.mutableState == nil {
 		response, err := c.getWorkflowExecutionWithRetry(ctx, &persistence.GetWorkflowExecutionRequest{
-			DomainID:  c.domainID,
-			Execution: c.workflowExecution,
+			DomainID:   c.domainID,
+			Execution:  c.workflowExecution,
+			DomainName: c.GetDomainName(),
```

Comment:
Should use `domainEntry.GetInfo().Name` here
c.GetDomainName() returns an empty domain name on cache failure

Reply:
Could you make the change?